pkg/controllers/management/kontainerdrivermetadata/data.go

```diff
@@ -450,9 +450,22 @@ func SaveSettings(k8sCurrVersions map[string]interface{},
 	if err := settings.KubernetesVersionToServiceOptions.Set(k8sSvcOptionData); err != nil {
 		return err
 	}
-	defaultK8sVersion, ok := rancherDefaultK8sVersions[rancherVersion]
-	if !ok || defaultK8sVersion == "" {
-		defaultK8sVersion = rancherDefaultK8sVersions["default"]
+	defaultK8sVersionRange, ok := rancherDefaultK8sVersions[rancherVersion]
+	if !ok || defaultK8sVersionRange == "" {
+		defaultK8sVersionRange = rancherDefaultK8sVersions["default"]
+	}
+	// get matching default k8s from k8s curr
+	toMatch := util.GetTagMajorVersion(defaultK8sVersionRange)
+	defaultK8sVersion := ""
+	for k8sCurr := range k8sCurrVersions {
```

Comment:
does UI pick the default from k8sCurrVersions or from KubernetesVersionToSystemImages?

Reply:
I believe the latter, but both have the same values